libvast/vast/chunk.hpp

```diff
@@ -98,10 +101,10 @@ class chunk : public caf::ref_counted,
 };
 
 /// @relates chunk
-bool operator==(const chunk& x, const chunk& y);
+/// @pre `x != nullptr`
+caf::error inspect(caf::serializer& sink, const chunk_ptr& x);
```

Comment:
This looks very hackish to me. A chunk is per definition unstructured and the `deleter` is silently dropped in the implementation. I would mark this deprecated and remove it as soon as we can use memory mapped files for persistence.

Reply:
The question is: does it matter during serialization? I would argue the semantics are actually fine: when you write out the block of memory, you can read it in independently of the deleter. This implies that a previously mmap'ed chunk may now be just a chunk on the heap. (If explicit mmap'ing is desired, then the user would go through the corresponding factory function.)

Reply:
My worry is that this opens the door for hard to debug problems. Image someone adds a pointer to a struct that is serialized in a chunk somewhere in the code. The change would still compile but we'd have strange bugs at runtime.